src/main/java/net/minecraftforge/event/entity/ProjectileImpactEvent.java

```diff
@@ -42,4 +47,41 @@ public Projectile getProjectile()
     {
         return projectile;
     }
+
+    /**
+     * {@inheritDoc}
+     * @deprecated Use {@link #setImpactResult(ImpactResult)} instead.
+     */
+    @Deprecated
+    public void setCanceled(boolean cancel)
+    {
+        super.setCanceled(cancel);
+    }
+
+    public void setImpactResult(@NotNull ImpactResult result)
+    {
+        this.result = Objects.requireNonNull(result, "ImpactResult cannot be null");
+    }
+
+    public ImpactResult getImpactResult()
+    {
+        return result;
+    }
+
+    public enum ImpactResult
+    {
+        /**
+         * The default vanilla action will be taken.
+         */
+        DEFAULT,
+        /**
+         * The projectile will continue flying as if the hit was not processed.
+         */
+        CONTINUE,
```

Comment:
I am confused about the implementation of this it is documented 'as if the hit was not processed' Im unsure what that means, does it mean that it just 'passes through' the entity and hits the next thing that it would of hit if the entity wasn't there? Right now the implementation makes the entity 'shield' all other entities behind it by breaking out of the entity hit loop. I would think that it would still allow a entity behind this one to be hit. 

To achieve this you'd have to manually add this entity to the ignored list, and increase the piecing ability of the arrow by 1.

I see a few states:

DEFAULT  - Behave as vanilla 
SKIP - Skips this entity as it if didn't exist, allowing other entities or blocks be to be hit behind it. This is what I would think 'CONTINUE' is.
STOP_PIERCING - Behave as vanilla but set piercing to 0. Allows entities to 'protect' entities and blocks behind them. 
Maybe a 'STOP'/'KILL' {or some better name} that acts like STOP_PIERCING but doesn't hurt the entity?